### ISSUE TITLE
CSharpLintBear does not give mcs all cs files together

### ISSUE THREAD

**Issue by jayvdb** [coala/coala-bears#1185]
CSharpLintBear is only given each file individually, which means it cant find other components in the same project.

This results in MAJOR errors like the following on [antonpup/Aurora](https;//github.com/antonpup/Aurora).

```
Project-Aurora/Profiles/Generic_Application/GenericApplicationSettings.cs
|   1| usingâ€¢Aurora.Settings;
|    | [MAJOR] CSharpLintBear:
|    | The type or namespace name `Settings' does not exist in the namespace `Aurora'. Are you missing an assembly reference?
```

This makes CSharpLintBear unusable for any serious C# project where the code is split across multiple files.

https://github.com/coala/bear-docs/blob/master/docs/CSharpLintBear.rst - it has no options at all.

**Comment by jayvdb** [comment]
As mcs is just a compiler, just running it on the files is unlikely to produce useful results.
I almost think this bear should be deleted, and replaced with proper static analysis tools which do exist in the mono project.

**Comment by jayvdb** [comment]
Another build shows [`'mono-winforms' is already installed.`](https://travis-ci.org/jayvdb/Aurora/builds/185754115#L194), but also [the log](https://clbin.com/5oL80) includes:

```
The type or namespace name 'Application' does not exist in the namespace 'System.Windows'. Are you missing an assembly reference?
```

So it isnt finding system libraries either.
Using `xbuild` is the better approach to getting compilation working, but that isnt the goal of a linter.

**Comment by Makman2** [comment]
That cries for a GlobalBear. We (probably I ... :/ ) should get global-bear support done for linter^^

**Comment by jayvdb** [comment]
fwiw, `VultureBear` is already a `GlobalBear` linter.

Maybe this bear can be made sort of useful by being a `GlobalBear`.

On my machine, running mcs on all .cs files in https://github.com/coala/coala-bears/issues/https;//github.com/antonpup/Aurora results in only

```
./Script_Examples/example_script.cs(9,14): error CS0101: The namespace `global::' already contains a definition for `ExampleEffect'
./Script_Examples/example_script_multi_layer.cs(10,14): (Location of the symbol related to previous error)
./Project-Aurora/Utils/ColorUtils.cs(235,13): error CS0227: Unsafe code requires the `unsafe' command line option to be specified
./Project-Aurora/Utils/BitmapUtils.cs(34,17): error CS0227: Unsafe code requires the `unsafe' command line option to be specified
./Project-Aurora/EffectsEngine/EffectLayer.cs(414,13): error CS0227: Unsafe code requires the `unsafe' command line option to be specified
./Project-Aurora/EffectsEngine/EffectLayer.cs(539,13): error CS0227: Unsafe code requires the `unsafe' command line option to be specified
./Project-Aurora/EffectsEngine/EffectLayer.cs(645,13): error CS0227: Unsafe code requires the `unsafe' command line option to be specified
```

The main reason this is so small is that compiling multiple files together turns off the linker.

This helps for some cases, but it is still not static analysis.  There are still problems for projects that require being compiled with `gmcs` or `smcs`.

**Comment by Makman2** [comment]
Okay so what do you suggest? It seems that making it a GlobalBear would be at least a step forward.

**Comment by jayvdb** [comment]
I'd suggest running `mcs **.cs` on Aurora in our docker image, to see if that is the same as my results above.  If it works like that, a `GlobalBear` is a good "quick win".
IMO it should also be renamed, as it isnt a conventional linter, and nobody is seriously using this yet, except on single .cs file projects, ... which isnt serious ;-)

**Comment by jayvdb** [comment]
Switching to `xbuild` would solve problems like https://github.com/antonpup/Aurora/pull/269 .
Whether we use `mcs **.cs` or `xbuild`, this is essentially a `BuildBear`, or at least a`CompileBear`.  Do we have any other bears which do compilation?

**Comment by Makman2** [comment]
Sounds good.

> [...] or at least a `CompileBear`

Then let's call it `CSharpSyntaxBear`, as the compiler only emits (usually) syntax problems.

> Do we have any other bears which do compilation?

not that I know of^^

**Comment by sils** [comment]
Yes @Makman2 we need that support desperately, there are already like 3 bears that do it manually and this one's next
